src/inc/regdisp.h

```diff
@@ -435,10 +435,20 @@ inline void FillRegDisplay(const PREGDISPLAY pRD, PT_CONTEXT pctx, PT_CONTEXT pC
 
     pRD->ctxPtrsOne.Lr = &pctx->Lr;
 #elif defined(_TARGET_X86_) // _TARGET_ARM_
+    pRD->pEdi = &(pctx->Edi);
```

Comment:
This won't work for the stack walking with WIN64EXCEPTIONS. When WIN64EXCEPTIONS are defined, the stack walking uses pCurrentContextPointers and doesn't know anything about the pEsi, pEbx etc. So while the unwinder would store the context pointers in these, the pCurrentContext would always just point to the context structure instead of the real location where the register value came from.
What I believe you need to do instead is to remove the setting of the pEsi etc here and then in the OOPStackUnwinderX86::VirtualUnwind set the pRD->pEsi etc to the corresponding values from the pCurrentContext before calling the CodeManager::UnwindStackFrame. And right after calling them, copy them from the RegDisplay::pEsi etc to the corresponding members of the pCurrentContext.
Also, you need to do that only for EDI, ESI, EBX and EBP. The CodeManager::UnwindStackFrame doesn't update any others. These are the only callee saved registers.

Also, the CopyRegDisplay needs to be modified so that it uses the code path that's now for non-x86 targets for x86 on Linux too.